### DISCUSSION TITLE
[RPi Adventures] - Is there a documentation somewhere about /usr?

### DISCUSSION THREAD

**Discussion by ArKam** [flatcar/Flatcar#1324] [Q&A] [answered]
Hi folks!

In my adventure on Flatcar + RPi setup, I'm looking for a way to optimize the /usr partition for the small machine, is there a documentation somewhere that explain how partitions like /usr are built out of these links?

https://www.flatcar.org/docs/latest/reference/developer-guides/sdk-building-production-images/
&
https://www.flatcar.org/docs/latest/reference/developer-guides/sdk-disk-partitions/

Especially the crypt volume part of it, is it a LUKS? A feature of BTRFS? (Not knowing btrfs super well).

As it would require some sort of cryptographic keys I guess and I'd like to know how they're handled and if I can swap the /usr partition from crypted volume to simple uncrypted volume as for RPi the cost of the crypt layer is having a real impact on responsiveness and performance.

Thanks in advance!

**Comment by pothos** [comment, marked as answer]
To build the `/usr` from scratch you would have to follow https://www.flatcar.org/docs/latest/reference/developer-guides/sdk-modifying-flatcar/ - the underlying technology is dm-verity, the crypto part is just for verification of integrity.
Building your own `/usr` means you won't consume updates anymore, though (basically, Flatcar consists of just `/usr` and the rest is established from there).

What is the goal you want to achieve?

**Reply by ArKam** [reply]
Alright! Thanks for the help, I’ll have to verify something on my own side as I suspect the lag issue to be on the network side of my lab as it also happens on default RaspiOS where it didn’t earlier.

But, in case it occurs my issue is indeed caused by BTRFS not being really suited for SDCard scenarios, what would you suggest me to replace it out of the « Craft your own version » solution?

BTW: Thank you very much for flatcar and your help, I love it xD

**Reply by pothos** [reply]
The `/usr` partition is read-only, so it doesn't matter if it's btrfs, squashfs or anything else - the SDCard won't be written to. For the rootfs the default is ext4 and you could switch to something like btrfs which I think is more suited to sdcard than ext4 because it also has special handling for SSDs and I'm not sure ext4 also has this.
